cmd/root_options.go

```diff
@@ -61,6 +63,14 @@ func (ro *RootOptions) Validate() []error {
 		}
 		return errArr
 	}
+
+	// check that the kernel versions supports at least one of probe and module
+	kr := kernelrelease.FromString(ro.KernelRelease)
```

Comment:
Can we do this check later, during the `toBuild` call, while we are already doing these checks?

Reply:
In `toBuild` it's slightly different: assuming that the options are valid, we try to fix them if they are not optimal (e.g. one of probe or kmod is known to be unfeasible)

Reply:
Yep i see :/ i don't know if making a new `builder.Validate()` api would help us?

Reply:
IIUC the Cmd modules individually call `rootOptions.toBuild()`, like so for the docker cmd:

```go
if !configOptions.DryRun {
    if err := driverbuilder.NewDockerBuildProcessor(viper.GetInt("timeout"), viper.GetString("proxy")).Start(rootOpts.toBuild()); err != nil {
        logger.WithError(err).Fatal("exiting")
    }
}
```

So, perhaps add `builder::Validate()`, but then call that from `cmd::toBuild()` and have `toBuild()` return `err` if it's invalid? That way we can leave existing callers as they are.

Reply:
Sure, the original solution by Jason is fine too! =)

I'm not that opinionated about this, since after all there are only 2 callers of `toBuild()`, so updating existing users isn't a big deal.